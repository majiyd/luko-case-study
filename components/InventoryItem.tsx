import React from "react";
import { View, Text, StyleSheet, Image } from "react-native";
import Colors from "../constants/Colors";
import { Inventory } from "../types";
import BaseText from "./BaseText";

type Props = {
  details: Inventory;
};
const InventoryItem = ({ details }: Props) => {
  const { name, purchasePrice, photo } = details;
  return (
    <View style={styles.item} testID={name}>
      <View style={styles.imageWrapper}>
        <Image source={{ uri: photo }} style={styles.image} />
      </View>
      <View style={styles.content}>
        <BaseText type="headline">{name}</BaseText>
        <BaseText style={styles.price}>
          â‚¬{purchasePrice.toLocaleString()}
        </BaseText>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  item: {
    flex: 1,
    width: 158,
    minHeight: 265,
    backgroundColor: Colors.light.background,
    marginBottom: 20,
    borderRadius: 14,

    shadowColor: Colors.light.shadow,
    shadowOffset: {
      width: 0,
      height: 10,
    },
    shadowOpacity: 0.12,
    shadowRadius: 20,
    elevation: 8,
  },
  imageWrapper: {
    width: "100%",
    aspectRatio: 1,
  },
  image: {
    flex: 1,
    resizeMode: "cover",
    borderTopLeftRadius: 14,
    borderTopRightRadius: 14,
  },
  content: {
    paddingVertical: 15,
    paddingHorizontal: 20,
    justifyContent: "space-between",
    flex: 1,
  },
  price: {
    marginTop: 5,
  },
});
export default InventoryItem;
