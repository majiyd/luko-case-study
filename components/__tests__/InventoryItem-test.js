import * as React from "react";
import renderer from "react-test-renderer";
import InventoryItem from "../InventoryItem";

const details = {
  id: 1,
  name: "Cartier ring",
  purchasePrice: 5780,
  type: "JEWELRY",
  description: "Gift from my grandfather",
  photo: "https://i.ibb.co/znXC7LQ/marcus-lewis-U63z-XX2f7ho-unsplash.jpg",
}

it(" InventoryItem renders correctly", () => {
  const tree = renderer.create(<InventoryItem details={details} />).toJSON();
  expect(tree).toMatchSnapshot();
});
