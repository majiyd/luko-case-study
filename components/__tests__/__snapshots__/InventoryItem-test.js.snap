// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` InventoryItem renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 14,
      "elevation": 8,
      "flex": 1,
      "marginBottom": 20,
      "minHeight": 265,
      "shadowColor": "rgb(6, 8, 13)",
      "shadowOffset": Object {
        "height": 10,
        "width": 0,
      },
      "shadowOpacity": 0.12,
      "shadowRadius": 20,
      "width": 158,
    }
  }
>
  <View
    style={
      Object {
        "aspectRatio": 1,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://i.ibb.co/znXC7LQ/marcus-lewis-U63z-XX2f7ho-unsplash.jpg",
        }
      }
      style={
        Object {
          "borderTopLeftRadius": 14,
          "borderTopRightRadius": 14,
          "flex": 1,
          "resizeMode": "cover",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2C2302",
            "fontSize": 19,
            "fontWeight": "bold",
            "lineHeight": 26,
          },
          undefined,
        ]
      }
    >
      Cartier ring
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#6B6651",
            "fontSize": 15,
            "lineHeight": 20,
          },
          Object {
            "marginTop": 5,
          },
        ]
      }
    >
      â‚¬
      5,780
    </Text>
  </View>
</View>
`;
